### PR TITLE
Feature/sc 6358/redis connection lost fixes

### PR DESCRIPTION
**Pull request by CeEv**
# Description

https://github.com/hpi-schul-cloud/node-notification-service/pull/250
Will fix the linter, path and extend class issues. Please only review the changes.

## Links to Tickets or other pull requests
- https://ticketsystem.schul-cloud.org/browse/SC-6358

## Approval for review

- [x] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### Link to Definiton of Done

More and detailed information on the _definition of done_ can be found [on Confluence](https://docs.schul-cloud.org/pages/viewpage.action?pageId=92831762)

### CODE REVIEW

File: src/routes/push.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ import Utils from '@/utils';
 
 const router: express.Router = express.Router();
 
-const pushService = new PushService();
+const pushService = new PushService('PushService');
```
[the remark this answers is not in the document]
Comment (tofixx) [reply src/routes/push.ts:10]: I just switched Codacy from ESLint to TSLint, so see if this has an effect, as the issue is not an issue. If this does not work, I may clean this up. @CeEv

File: src/services/BaseService.ts [hunk 2]
```diff
@@ -152,36 +145,112 @@ export default abstract class BaseService {
...
+				done(null)
+			} else if (error.responseCode === 421 && error.message.includes('421 Rate limit reached. Please try again later')) {
+				// TODO: eskalation send email to admin do not work at this position (?)
+				// Send to sentry
+				escalation('Rate limit reached, it is paused for ' + intervallTime, error);
+				this.pausedQueue(intervallTime);
+				done(error);
+			} else if (error.responseCode === 421 && error.message.includes('421 Reject due to policy violations')) {
+				// -> eskalation Sentry
+				escalation('Our email account is blocked, please contact iones.', error);
```
Comment (Metauriel) [inline src/services/BaseService.ts:203]: Ionos

**Review by Metauriel — DISMISSED** [review]
obviously linter has to be fixed, and I wonder why did you remove the region comments for the class definitons?
Otherwise it definetly looks like an improvement :)

**Review by Metauriel — APPROVED** [review]

File: src/services/BaseService.ts (hunk 2 above)
Comment (CeEv) [reply src/services/BaseService.ts]: done

File: platforms/config.default.json [hunk 3]
```diff
@@ -37,13 +37,17 @@
...
+    "retries": 6,
+    "backoffTime": 120000,
+    "backoffStrategy": "fixed",
     "defaults": {
       "prefix": "notification",
       "stallInterval": 5000,
       "nearTermWindow": 1200000,
       "delayedDebounce": 1000,
       "redis": {
+        "host": "127.0.0.1",
```
Comment (alriese) [inline platforms/config.default.json:49]: Is localhost a good default address?

File: src/helper/logger.ts [hunk 4]
```diff
@@ -36,8 +48,8 @@ export class LoggerStream {
 	}
 }
 
-if (process.env.NODE_ENV === 'production') {
-	logger.debug('Logging initialized at production level');
+if (isProductionMode) {
+	console.log('Logging initialized at production level');
```
Comment (alriese) [inline src/helper/logger.ts:52]: Why `console.log` instead of the logger?

File: src/services/BaseService.ts [hunk 5]
```diff
@@ -12,7 +15,26 @@ function getType(object: object | null) {
...
 
 export default abstract class BaseService {
-	// region public static methods
+
+	private static queues: Queue[] = [];
+	private transporters: PlatformTransporter[] = [];
+	private name: string = 'NoName';
+
+	protected constructor(name: string) {
+		this.name = name;
```
Comment (tofixx) [inline src/services/BaseService.ts:24]: If using only PushService or MailService as name, this._serviceType() will respond a similar value which is defined in https://github.com/hpi-schul-cloud/node-notification-service/blob/9813ae91659a1ebcec8da593e221c2ef7e08e61e/src/services/BaseService.ts#L142 and overridden in https://github.com/hpi-schul-cloud/node-notification-service/blob/a605005c89f5f3134523a893946adca636bf81e4/src/services/MailService.ts#L69 and https://github.com/hpi-schul-cloud/node-notification-service/blob/a605005c89f5f3134523a893946adca636bf81e4/src/services/PushService.ts#L57

File: platforms/config.default.json (hunk 3 above)
Comment (CeEv) [reply platforms/config.default.json:49]: It is also used in live system and it is implicit set with the same value. I only add it to show that we can change it at this position.

File: src/helper/logger.ts (hunk 4 above)
Comment (CeEv) [reply src/helper/logger.ts:52]: The logger is not defined on this position.

File: src/services/BaseService.ts (hunk 5 above)
Comment (CeEv) [reply src/services/BaseService.ts:24]: This is true. I see it later. But not sure it it is the why that we should go. 
In escalation logic i add a other name for mail server like for the base mail service to see which redis connection and query is used, or throw an error.
In future we must outsource the job-query logic from BaseService that we only have one job-query and one redis connection to solve the read SCAN/write errors. If this is done we can remove the name attribute if we want.

File: src/helper/logger.ts (hunk 4 above)
Comment (tofixx) [reply src/helper/logger.ts:52]: check 👍   but can https://github.com/hpi-schul-cloud/node-notification-service/blob/43f2a4d537970a5b9adddd30af660c9d24d6cbd2/src/app.ts#L57 use the logger

**Pull request merged**